app_mesh.tf

```
@@ -0,0 +1,203 @@
+data "aws_acmpca_certificate_authority" "root_ca" {
+  arn = "arn:aws:acm-pca:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:certificate-authority/1b3cd270-9086-4c72-bfee-e8d308439481"
+}
+data "aws_acm_certificate" "internal" {
+  domain   = "*.apps.local"
+  statuses = ["ISSUED"]
+}
+locals {
+  health_check_proto = lower(lookup(var.health_check, "protocol", "grpc"))
+  app_mesh_proto     = contains(["http", "https"], local.health_check_proto) ? "http" : "grpc"
+}
+resource "aws_appmesh_virtual_node" "this" {
+  count     = var.app_mesh_configuration.enabled ? 1 : 0
+  mesh_name = local.mesh_name
+  name      = var.service_name
+  spec {
+    dynamic backend {
+      for_each = var.app_mesh_configuration.backends
+      content {
+        virtual_service {
+          virtual_service_name = backend.value
+          dynamic client_policy {
+            ## activate tls context policy only when this service is within our domain
+            for_each = length(regexall("\\.services\\.vpc\\.internal$", backend.value)) > 0 ? [1] : []
+            content {
+              tls {
+                validation {
+                  trust {
+                    acm {
+                      certificate_authority_arns = [
+                        data.aws_acmpca_certificate_authority.root_ca.arn
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    listener {
+      health_check {
+        protocol            = local.app_mesh_proto
+        path                = lookup(var.health_check, "path", null)
+        healthy_threshold   = 2
+        unhealthy_threshold = 2
+        timeout_millis      = 2000
+        interval_millis     = 5000
+      }
+      port_mapping {
+        port     = var.container_port
+        protocol = local.app_mesh_proto
+      }
+      tls {
+        certificate {
+          acm {
+            certificate_arn = data.aws_acm_certificate.internal.arn
+          }
+        }
+        mode = "STRICT"
+      }
+    }
+    logging {
+      access_log {
+        file {
+          path = "/dev/stdout"
+        }
+      }
+    }
+    service_discovery {
+      aws_cloud_map {
+        namespace_name = "apps.local"
+        service_name   = var.service_name
+      }
+    }
+  }
+}
+
+resource "aws_appmesh_route" "grpc" {
+  for_each = lookup(var.app_mesh_configuration, "grpc_endpoints", {})
```

```suggestion
  for_each = var.app_mesh_configuration.enabled ? lookup(var.app_mesh_configuration, "grpc_endpoints", {}) : {}
```

to support usages without an app mesh